### ISSUE TITLE
Windows downgrade fails to remove data

### ISSUE THREAD

**Issue by mook-as** [rancher-sandbox/rancher-desktop#712]
When downgrading on Windows, we attempt to delete the data on the data WSL distribution; however, if the cluster (kubelet) was previously running, it creates a bunch of tmpfs mounts, blocking the deletion:

```
rm: can't remove '/var/lib/kubelet/pods/1938ac0a-2045-4c3e-aba2-9711982ead31/volumes/kubernetes.io~projected/kube-api-access-wc9b9': Resource busy
rm: can't remove '/var/lib/kubelet/pods/21e88a7a-e320-4fc2-a0b2-f847be1b7386/volumes/kubernetes.io~projected/kube-api-access-6pxcm': Resource busy
rm: can't remove '/var/lib/kubelet/pods/43b82490-bdd7-473d-a45e-4895e37eccba/volumes/kubernetes.io~secret/certs-server': Resource busy
rm: can't remove '/var/lib/kubelet/pods/43b82490-bdd7-473d-a45e-4895e37eccba/volumes/kubernetes.io~secret/certs-ca': Resource busy
rm: can't remove '/var/lib/kubelet/pods/43b82490-bdd7-473d-a45e-4895e37eccba/volumes/kubernetes.io~projected/kube-api-access-vj8mv': Resource busy
rm: can't remove '/var/lib/kubelet/pods/46434c39-6ccf-47bf-9ee5-97b08ce77f4a/volumes/kubernetes.io~projected/kube-api-access-5nhtb': Resource busy
rm: can't remove '/var/lib/kubelet/pods/599c74ff-6c0f-42f2-909e-e08471a86099/volumes/kubernetes.io~projected/kube-api-access-tnclj': Resource busy
rm: can't remove '/var/lib/kubelet/pods/2d8ae630-5c6e-4d5e-a52b-b95ef9cdabe9/volumes/kubernetes.io~projected/kube-api-access-gl6bl': Resource busy
WSL failed to execute rm -rf /var/lib/kubelet
```

_Originally posted by @mattfarina in https://github.com/rancher-sandbox/rancher-desktop/pull/704#discussion_r719744583_

**Comment by mook-as** [comment]
To test:
- Start a cluster (e.g. at `v1.20.4`)
- Watch for things to be up (`cat /proc/self/mountinfo` should include mounts about `kubelet`)
- Downgrade the cluster (e.g. to `v1.20.0`)
- Watch for failure

Analysis:
- Kubelet creates a bunch of tmpfs mounts.
- These are on `/var/lib/kubelet` in the `rancher desktop` distribution, which is a bind mount to `/mnt/wsl/rancher-desktop/run/data/var/lib` (in the root mount namespace that we can't get a pid into), which is a bind mount to `/var/lib/kubelet` on the `rancher-desktop-data` distribution.
- We terminate k3s (and therefore kubelet).
- `/mnt/wsl/rancher-desktop/run/data` still exists (because it's in the root mount namespace)
- We don't have anything that can access the kubelet mount namespace anymore, so we can't unmount the various tmpfs.

**Comment by jandubois** [comment]
This is not fixed in the 0.6.0 release:

```
Registered distributions: Ubuntu-20.04,docker-desktop-data,openSUSE-Leap-15.2,Arch,docker-desktop,rancher-desktop,rancher-desktop-data
Could not use saved version v1.22.2, not in v1.21.5,v1.21.4,v1.21.3,v1.21.2,v1.21.1,v1.21.0,v1.20.11,v1.20.10,v1.20.9,v1.20.8,v1.20.7,v1.20.6,v1.20.5,v1.20.
Registered distributions: Ubuntu-20.04,docker-desktop-data,openSUSE-Leap-15.2,Arch,docker-desktop,rancher-desktop,rancher-desktop-data
Registered distributions: Ubuntu-20.04,docker-desktop-data,openSUSE-Leap-15.2,Arch,docker-desktop,rancher-desktop,rancher-desktop-data
Registered distributions: Ubuntu-20.04,docker-desktop-data,openSUSE-Leap-15.2,Arch,docker-desktop,rancher-desktop,rancher-desktop-data
data distro already registered
Found a valid mount with /dev/sdc: 179 178 8:32 / /mnt/wsl/rancher-desktop/run/data rw,relatime shared:3 - ext4 /dev/sdc rw,discard,errors=remount-ro,data=o
Installing C:\Users\Jan\AppData\Local\Programs\Rancher Desktop\resources\resources\linux\bin\trivy as /mnt/c/Users/Jan/AppData/Local/Programs/Rancher Deskto
Installing C:\Users\Jan\AppData\Local\Programs\Rancher Desktop\resources\resources\templates\trivy.tpl as /mnt/c/Users/Jan/AppData/Local/Programs/Rancher De
Deleting incompatible Kubernetes state due to downgrade from v1.22.2 to v1.21.5...
rm: can't remove '/var/lib/kubelet/pods/efce2c01-d923-4804-a61f-cbd233e04841/volumes/kubernetes.io~projected/kube-api-access-tzm4q': Resource busy
rm: can't remove '/var/lib/kubelet/pods/021b7785-ce18-4c81-a5c1-e420e8a8b1db/volumes/kubernetes.io~projected/kube-api-access-5nk25': Resource busy
rm: can't remove '/var/lib/kubelet/pods/0d755748-77e3-40a7-ab19-ae48ce4d84f9/volumes/kubernetes.io~projected/kube-api-access-9txld': Resource busy
rm: can't remove '/var/lib/kubelet/pods/df87bf24-1018-41fd-9626-77f41f1ac630/volumes/kubernetes.io~projected/kube-api-access-hkb8n': Resource busy
rm: can't remove '/var/lib/kubelet/pods/0b573b10-0d28-4a8d-a1bb-6b3094a87531/volumes/kubernetes.io~secret/certs-server': Resource busy
rm: can't remove '/var/lib/kubelet/pods/0b573b10-0d28-4a8d-a1bb-6b3094a87531/volumes/kubernetes.io~secret/certs-ca': Resource busy
rm: can't remove '/var/lib/kubelet/pods/0b573b10-0d28-4a8d-a1bb-6b3094a87531/volumes/kubernetes.io~projected/kube-api-access-g2285': Resource busy
WSL: executing: rm -rf /var/lib/kubelet: Error: wsl.exe exited with code 1
```

Rebooting the machine did resolve the problem as expected.

**Comment by mook-as** [comment]
Moving to Done again; please reopen if we could still reproduce this. (Please attach enough logging / process listings / etc. to be able to indicate _why_ there's still an issue deleting, if that is the case.)
